File: backend-laravel/app/Models/Entrega.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Entrega extends Model
{
    protected $fillable = [
        'tarea_id',
        'user_id',
        'comentario_estudiante',
        'archivo_url',
        'calificacion',
        'comentario_profesor',
        'fecha_entrega',
    ];

    public function tarea(): BelongsTo
    {
        return $this->belongsTo(Tarea::class);
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}

File: app-aula/lib/models/entrega.dart
class Entrega {
  final int? id;
  final int tareaId;
  final int userId;
  final String? comentarioEstudiante;
  final String? archivoUrl;
  final double? calificacion;
  final String? comentarioProfesor;
  final String? fechaEntrega;

  Entrega({
    this.id,
    required this.tareaId,
    required this.userId,
    this.comentarioEstudiante,
    this.archivoUrl,
    this.calificacion,
    this.comentarioProfesor,
    this.fechaEntrega,
  });

  factory Entrega.fromJson(Map<String, dynamic> json) {
    return Entrega(
      id: json['id'],
      tareaId: json['tarea_id'],
      userId: json['user_id'],
      comentarioEstudiante: json['comentario_estudiante'],
      archivoUrl: json['archivo_url'],
      calificacion: json['calificacion'] != null ? double.parse(json['calificacion'].toString()) : null,
      comentarioProfesor: json['comentario_profesor'],
      fechaEntrega: json['fecha_entrega'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'tarea_id': tareaId,
      'user_id': userId,
      'comentario_estudiante': comentarioEstudiante,
      'archivo_url': archivoUrl,
      'calificacion': calificacion,
      'comentario_profesor': comentarioProfesor,
      'fecha_entrega': fechaEntrega,
    };
  }
}
